File: app/src/main/java/com/tydeya/familycircle/ui/conversationpart/MainConversationPage.java
package com.tydeya.familycircle.ui.conversationpart;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.tydeya.familycircle.App;
import com.tydeya.familycircle.R;
import com.tydeya.familycircle.data.conversationsinteractor.abstraction.ConversationInteractorCallback;
import com.tydeya.familycircle.data.conversationsinteractor.details.ConversationInteractor;
import com.tydeya.familycircle.ui.conversationpart.chatpart.MessagingActivity;

import javax.inject.Inject;

public class MainConversationPage extends Fragment implements MainConversationRecyclerViewAdapter.OnClickConversationListener, ConversationInteractorCallback {

    private RecyclerView recyclerView;
    private NavController navController;
    private MainConversationRecyclerViewAdapter recyclerViewAdapter;

    @Inject
    ConversationInteractor conversationInteractor;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        App.getComponent().injectFragment(this);

        View root = inflater.inflate(R.layout.fragment_main_conversation_page, container, false);
        recyclerView = root.findViewById(R.id.main_conversation_page_recycler_view);
        navController = NavHostFragment.findNavController(this);

        return root;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        conversationInteractor.subscribe(this);

        recyclerViewAdapter = new MainConversationRecyclerViewAdapter(getContext(),
                conversationInteractor.getConversations(), this);
        recyclerView.setAdapter(recyclerViewAdapter);

        recyclerView.setLayoutManager(new LinearLayoutManager(getContext(),
                RecyclerView.VERTICAL, false));
    }

    @Override
    public void onClickConversation(int position) {
        Intent intent = new Intent(getContext(), MessagingActivity.class);
        intent.putExtra("correspondencePosition", position);
        startActivity(intent);

    }

    @Override
    public void conversationsDataUpdated() {
        recyclerViewAdapter.refreshData(conversationInteractor.getConversations());
    }

    @Override
    public void onResume() {
        super.onResume();
        conversationInteractor.subscribe(this);
    }

    @Override
    public void onPause() {
        super.onPause();
        conversationInteractor.subscribe(this);
    }
}

File: app/src/main/java/com/tydeya/familycircle/ui/conversationpart/main/MainConversationPage.kt
package com.tydeya.familycircle.ui.conversationpart.main

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.tydeya.familycircle.App
import com.tydeya.familycircle.R
import com.tydeya.familycircle.domain.messenger.interactor.abstraction.MessengerInteractorCallback
import com.tydeya.familycircle.domain.messenger.interactor.details.MessengerInteractor
import com.tydeya.familycircle.ui.conversationpart.inconversation.InConversationActivity
import com.tydeya.familycircle.ui.conversationpart.main.createconversation.CreateConversationDialog
import com.tydeya.familycircle.ui.conversationpart.main.recyclerview.MainConversationRecyclerViewAdapter
import com.tydeya.familycircle.ui.conversationpart.main.recyclerview.MainConversationRecyclerViewOnClickListener
import kotlinx.android.synthetic.main.fragment_main_conversation_page.*
import javax.inject.Inject

class MainConversationPage
    :
        Fragment(R.layout.fragment_main_conversation_page),
        MessengerInteractorCallback,
        MainConversationRecyclerViewOnClickListener {

    @Inject
    lateinit var messengerInteractor: MessengerInteractor

    private lateinit var adapter: MainConversationRecyclerViewAdapter

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        App.getComponent().injectFragment(this)
        setAdapter()
        setCreateButton()
    }

    private fun setAdapter() {
        adapter = MainConversationRecyclerViewAdapter(context!!, ArrayList(), this)
        main_conversation_page_recycler_view.layoutManager =
                LinearLayoutManager(context!!, LinearLayoutManager.VERTICAL, false)
        main_conversation_page_recycler_view.adapter = adapter

    }

    private fun setCreateButton() {
        main_conversation_toolbar_create_new_conversation.setOnClickListener {
            val dialog = CreateConversationDialog()
            dialog.show(parentFragmentManager, "createConversationDialog")
        }
    }

    private fun setCurrentData() {
        adapter.refreshData(messengerInteractor.conversations)
    }

    override fun onConversationClick(conversationId: String) {
        messengerInteractor.actualConversationId = conversationId
        startActivity(Intent(context, InConversationActivity::class.java))
    }

    /**
     * Callbacks
     * */

    override fun messengerDataFromServerUpdated() {
        setCurrentData()
    }

    override fun onResume() {
        super.onResume()
        messengerInteractor.subscribe(this)
    }

    override fun onPause() {
        super.onPause()
        messengerInteractor.unsubscribe(this)
    }

}